dat.py

```diff
@@ -18,25 +21,29 @@ def __init__(self, filename):
         self.gene_index = {}
         for i in range(len(self.gene_list)):
             self.gene_index[self.gene_list[i]] = i
+        logger.debug("Got %s genes.", len(self.gene_list))
 
     def open_file(self, filename):
+        logger.debug("Opening %s", filename)
         dab_file = open(filename, 'rb')
 
         # get number of genes
         a = array.array('I')
         a.fromfile(dab_file, 1)
         size = a[0]
+        logger.debug("Expecting %s genes.", size)
 
         # get gene names
         start = 4
         end = 4
         count = 0
-        while count < a[0]:
+        while count < size:
             dab_file.seek(end)
-            if(dab_file.read(2) == '\0\0'):
+            chrs = dab_file.read(2)
+            if(chrs == b'\x00\x00'):
```

Comment:
Nice this should fix this issue https://github.com/FunctionLab/flib/pull/1#issuecomment-249849334 (I think)